smack-extensions/src/main/java/org/jivesoftware/smackx/xdata/provider/DataFormProvider.java

```diff
@@ -220,6 +220,9 @@ private static FormField parseField(XmlPullParser parser, XmlEnvironment xmlEnvi
                     // As per XEP-0004, text-single is the default form field type, which we use as emergency fallback here.
                     type = FormField.Type.text_single;
                 }
+            } else {
```

Comment:
I am sorry, but we where both wrong, this should be `if (formType != null) {`, then lines 220-221 can be removed.

Reply:
I don't think I understand. If the type is `null` we should do the following:

1. if its a FORM_TYPE field we set the type to hidden (this might not be completely correct, the XEP-0068 describes this behavior for forms of type `submit`, for `result` and `form` it writes: "If the FORM_TYPE field is not hidden in a form with type="form" or type="result", it MUST be ignored as a context indicator.")
2. if a FORM_TYPE was set, we lookup the field type in the FormFieldRegistry
3. if still no type is known, we use the default described in XEP-0004 `text-single`.

I think the current code implements this?

Reply:
Sorry, I wrote `formType`, when I meant `type` (the field's type).

```
            } else if (formType != null) {
￼                // If no field type was explicitly provided, then we need to lookup the
￼                // field's type in the registry.
￼                type = FormFieldRegistry.lookup(formType, fieldName);
                if (type == null) {
                    LOGGER.warning("The Field '" + fieldName + "' from FORM_TYPE '" + formType
                                    + "' is not registered. Field type is unknown, assuming text-single.");
￼                    // As per XEP-0004, text-single is the default form field type, which we use as emergency fallback here.
￼                    type = FormField.Type.text_single;
￼                }
￼            } else {
               // As per XEP-0004, text-single is the default form field type
￼               type = FormField.Type.text_single;
￼            }
```

should be

```
            } else if (formType != null) {
￼                // If no field type was explicitly provided, then we need to lookup the
￼                // field's type in the registry.
￼                type = FormFieldRegistry.lookup(formType, fieldName);
                if (type == null) {
                    LOGGER.warning("The Field '" + fieldName + "' from FORM_TYPE '" + formType
                                    + "' is not registered. Field type is unknown, assuming text-single.");
￼                }
￼            }

            if (type != null) {
               // As per XEP-0004, text-single is the default form field type
￼               type = FormField.Type.text_single;
￼            }